### PR TITLE
UCP/PROTO: Add common protocol initialization and info print

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@brminich can you pls review?

File: src/ucp/proto/proto_select.c [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+        status = UCS_ERR_NO_ELEM;
+        goto err_free_priv;
+    }
+
+    /* Finalize the shared priv buffer size */
+    if (offset == 0) {
+        ucs_free(proto_init->priv_buf);
+        proto_init->priv_buf = NULL;
+    } else {
+        tmp = ucs_realloc(proto_init->priv_buf, offset, "ucp_proto_priv");
```
Comment (brminich) [inline src/ucp/proto/proto_select.c:104]: does protocol keep an address to priv space (as it was passed to init function by `ucp_proto_id_call`)? This address can be changed by realloc

File: src/ucp/proto/proto_select.c [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    proto_init->priv_buf = ucs_malloc(ucp_protocols_count * UCP_PROTO_PRIV_MAX,
+                                      "ucp_proto_priv");
+    if (proto_init->priv_buf == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto err;
+    }
+
+    offset = 0;
+    for (proto_id = 0; proto_id < ucp_protocols_count; ++proto_id) {
+        init_params.priv      = UCS_PTR_BYTE_OFFSET(proto_init->priv_buf,
```
Comment (brminich) [inline src/ucp/proto/proto_select.c:74]: how is this space used by protocol?

File: src/ucp/proto/proto_select.c (hunk 1 above)
Comment (yosefe) [reply src/ucp/proto/proto_select.c:104]: (discussed offline) the pointer is assigned at later stage which did not fit into this PR

File: src/ucp/proto/proto_select.c (hunk 2 above)
Comment (yosefe) [reply src/ucp/proto/proto_select.c:74]: (discussed offline) the protocol can keep specific data there such as list of lanes, max_frag for each lane, etc.

**Pull request merged**